Tests/MongoSwiftTests/DocumentTests.swift

```diff
@@ -423,67 +464,86 @@ final class DocumentTests: XCTestCase {
             pointer = doc.data
         }
 
-        expect(doc).to(equal(["string": "hi", "nil": nil, "doc": newDoc, "arr": [3, 4] as [Int], "oid": oid]))
+        expect(doc).to(equal(["string": "hi", "nil": nil, "doc": newDoc, "arr": [3, 4] as [Int]]))
     }
 
     // test replacing both overwritable and nonoverwritable values with values of different types
     func testReplaceValueWithNewType() throws {
         // make a deep copy so we start off with uniquely referenced storage
-        var doc1 = Document(fromPointer: DocumentTests.overwritables.data)
+        var overwritableDoc = Document(fromPointer: DocumentTests.overwritables.data)
 
         // save a reference to original bson_t so we can verify it changes
-        var pointer1 = doc1.data
-
-        let newPairs1: [(String, BSONValue?)] = [("double", Int(10)), ("int32", "hi"), ("int64", Decimal128("1.0")), ("bool", [1, 2, 3]), ("decimal", 100)]
+        var overwritablePointer = overwritableDoc.data
+
+        let newOid = ObjectId()
+        let overwritablePairs: [(String, BSONValue?)] = [
+            ("double", Int(10)),
+            ("int32", "hi"),
+            ("int64", Decimal128("1.0")),
+            ("bool", [1, 2, 3]),
+            ("decimal", 100),
+            ("oid", 25.5),
+            ("timestamp", newOid),
+            ("datetime", Timestamp(timestamp: 1, inc: 2))
+        ]
 
-        newPairs1.forEach { (k, v) in
-            doc1[k] = v
-            expect(doc1.data).toNot(equal(pointer1))
-            pointer1 = doc1.data
+        overwritablePairs.forEach { (k, v) in
+            overwritableDoc[k] = v
+            expect(overwritableDoc.data).toNot(equal(overwritablePointer))
+            overwritablePointer = overwritableDoc.data
```

Comment:
:+1: for more descriptive variables names.

Reply:
that's on me 😛